product_archive_sold_product/models/product_product.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _process_product_archive(self):
+        self.env.cr.execute(
+            """
+            SELECT
+                product_id,
+                sum(qty_delivered) AS sale_qty
+            FROM
+                sale_order_line
+            JOIN
+                product_product ON product_product.id = sale_order_line.product_id
+            WHERE
+                state in ('sale', 'done') AND active = True
+            GROUP BY
+                product_id
+        """
+        )
+
+        sale_dict = self.env.cr.dictfetchall()
+
+        sale_vals = {}
+        for dictionaly in sale_dict:
+            sale_vals[dictionaly["product_id"]] = dictionaly["sale_qty"]
```

Comment:
```suggestion
        prod_ids = [r[0] for r in self.env.cr.fetchall()]
```
`prod_ids = self.env.cr.fetchall()` might just work as well, but I don't feel comfortable with throwing a list of tuples in `browse()`.